### DISCUSSION TITLE
Independence of layers

### DISCUSSION THREAD

**Discussion by mausch** [milvus-io/milvus#19302] [Q&A and General discussion] [answered]
Hi, I just updated the image of an existing Milvus install from v2.1.1 to a nightly image tag.
The IndexCoord started crashing with a panic "runtime error: invalid memory address or nil pointer dereference".

My question is about how this affected other layers/components of Milvus.
My expectation was that in this state I would not be able to add new vectors to a index but I would still be able to query existing vectors.
[The architecture documentation says "These layers are mutually independent when it comes to scaling or disaster recovery."](https://milvus.io/docs/architecture_overview.md)
However it seems that IndexCoord crashing made pretty much everything else crash too?

The proxy crashes with `error="find no available indexcoord, check indexcoord state"` which is to be expected.
But QueryCoord also crashes with the same error, which is unexpected to me.

So my questions are:

1. Is it to be expected that a failure in Indexing crashes other components?
2. If so, are there plans to make querying resilient to this kind of failures?

Many thanks

**Comment by yhmo** [comment, marked as answer]
We also noticed this problem. QueryCoord should not be blocked when IndexCoord crashed. We have made a fix on the master branch, to make the coordinators work independently. The fix is available in v2.2.

And we plan to make a design change after v2.2: combine all the coordinators into one coordinator, to reduce the system complexity.
